common/network-shuffle/src/test/java/org/apache/spark/network/shuffle/RemoteBlockPushResolverSuite.java

```diff
@@ -257,6 +274,7 @@ public void testDuplicateBlocksAreIgnoredWhenPrevStreamIsInProgress() throws IOE
     pushResolver.finalizeShuffleMerge(new FinalizeShuffleMerge(TEST_APP, NO_ATTEMPT_ID, 0, 0));
     MergedBlockMeta blockMeta = pushResolver.getMergedBlockMeta(TEST_APP, 0, 0, 0);
     validateChunks(TEST_APP, 0, 0, 0, blockMeta, new int[]{4}, new int[][]{{0}});
+    verifyMetrics(4, 0, 0, 0, 0, 0);
```

Comment:
We should be capturing this in metrics - essentially streams which were ignored due to a speculative task writing concurrently.
We can add it as a follow up though.

Reply:
> We should be capturing this in metrics

Not sure if I'm following here, could you please elaborate? Thanks @mridulm !

Reply:
The specific testcase here is, we have duplicate blocks being sent (via speculative execution for example), and so we want to ignore data from the "newer" push. For example, here we stream2 would be ignored. The metrics are not capturing it - and so we see the same result in metrics as what we see for a successful push (what we would see for `testBasicBlockMerge`).

Reply:
Thanks @mridulm. So you are suggesting to add a new metric to capture size/byte of pushed blocks that got ignored, right? If so, I'm trying to figure out what insights we could get if we add number/size of ignored blocks, thoughts? ++ @otterc

Reply:
Sounds good, added a new metric `ignoredBlockBytes` to tracks the size of the blocks that are ignored. @mridulm @otterc